src/Silakov.Homework8/PeopleGenerator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading;
+
+namespace Silakov.Homework8
+{
+    public class PeopleGenerator
+    {
+        private static ThreadLocal<Random> _random;
+        private static string[] _predefinedNames;
+
+        public PeopleGenerator()
+        {
+            var seed = Environment.TickCount;
+            _random = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref seed)));
```

Comment:
Please describe what this initialization was intended to achieve

Reply:
I read on one of the sites that the Random is not thread-safe. Since I am using the Random constructor without parameters, it uses the current date to generate a random value. As a parameter, I pass the number of milliseconds after the start of the system (Environment.TickCount). Interlocked.Increment must ensure that the value is not incremented not simultaneously. 
I spied a ready-made solution. Did I overcomplicated the Person generator?

Reply:
Yes, besides the complication, there are some mistakes that correspond to the understanding of multithreading:
1. var seed = Environment.TickCount; - here you are copying the tick count value to the local variable that is different on each thread (because each thread have own stack)
2. Interlocked.Increment(ref seed) - here you use the synchronization for closure variable, it can lead to some unpredicted behavior. In the example you found they try to synchronize here to avoid situation when Random object is created in nere same time by different threads that can cause the same "random" values.
3. What is not thread safe in Random is the .Next() method, it can cause the case when multiple threads get the same "random" value. By that reason the author of example you found tries to have own instance of random for each thread. But basically you are not calling .Next() from different threads, so there is no need for this improvement.

Reply:
Thank you very much for the detailed answer